### ISSUE TITLE
Support name matching on variations

### ISSUE THREAD

**Issue by ajf58** [camdram/camdram#341]
This is a feature request taken from https://www.camdram.net/admin/issues/76042, where the user would like to match names against common variations. Fore example Tom Smith and Thomas Smith would return the same search result, and would be considered to be the same person throughout the site.

I've labelled this as an enhancement and put it on a future milestone. I think there's a significant level of effort required to implement this.

**Comment by philosophicles** [comment]
I'm not sure this, as stated, is quite what the support ticket was after.

The example in the support ticket was more like matching Ben Smith to
Benjamin Smith. This is subtly different because 'Ben' are the first three
characters in both strings, whereas 'Tom' and 'Tho' have differed already.

In SQL terms 'Ben% Smith' matches both, whereas 'Tom% Smith' doesn't.

So one option could be to solve the potentially simpler case only? The
level of effort would still be significant though, and increases the chance
of confusing two genuinely different people. Totally agree this is quite a
far future thing.

For the "full complexity" case, string distance algorithms like
Levenshtein-Damerau or Jaro-Winkler will possibly be involved. I've used
these before, false positives and processing time can be problematic.
Soundex/metaphone algorithms may also work.

I will say that most enterprise level software I've seen shies away from
trying to implement such things (certainly without human review),
preferring to leave it in human hands...

A totally alternative approach would be to move further towards unifying
camdram's person and user entities, eg involving email addresses somehow
(privately) in person entries, which can then be used as a unique ID. (This
has been discussed elsewhere, some time ago.)

**Comment by ajf58** [comment]
Yes you're right, for the specific example it's a Ben versus Benjamin thing, but presumably don't match against Benedict. It's certainly a very complicated problem to solve, and low priority.
